src/Service/Search/SolrSuggest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Search\Service\Search;
+
+use Atoolo\Search\Dto\Search\Query\SuggestQuery;
+use Atoolo\Search\Dto\Search\Result\Suggestion;
+use Atoolo\Search\Dto\Search\Result\SuggestResult;
+use Atoolo\Search\Exception\UnexpectedResultException;
+use Atoolo\Search\Service\SolrClientFactory;
+use Atoolo\Search\SuggestSearcher;
+use JsonException;
+use Solarium\Core\Client\Client;
+use Solarium\QueryType\Select\Query\Query as SolrSelectQuery;
+use Solarium\QueryType\Select\Result\Result as SolrSelectResult;
+
+/**
+ *  @phpstan-type SolariumResponse array{
+ *      facet_counts: array{
+ *         facet_fields:array<string,array<string>>
+ *      }
+ *  }
+ *
+ * Implementation of the "suggest search" based on a Solr index.
+ */
```

Comment:
let's put the phpdoc annotations at the end of comments:

```suggestion
/**
 * Implementation of the "suggest search" based on a Solr index.
 *
 *  @phpstan-type SolariumResponse array{
 *      facet_counts: array{
 *         facet_fields: array<string, array<string>>
 *      }
 *  }
 */
```